### PR TITLE
Descending into nested groups

### PR DESCRIPTION
**Pull request by mikolak**
User is able to descend into nested group (no top/back navigation or saving navigation state)

### CODE REVIEW

**Comment by mikolak** [comment]
TODO: change new adapter's package

File: cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/AdapterModel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.polidea.cockpit.core.CockpitParam
+import com.polidea.cockpit.core.CockpitParamGroup
+
+internal class DisplayModel(val items: List<DisplayItem>)
+
+internal sealed class DisplayItem(val displayName: String?) {
+    class Section(displayName: String?) : DisplayItem(displayName)
+    class Group(displayName: String?, val group: CockpitParamGroup) : DisplayItem(displayName)
+    class Param(displayName: String?, val param: CockpitParam<Any>) : DisplayItem(displayName)
+//    class TopNavigation : DisplayItem(null)
```
Comment (martawoldanska) [inline cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/AdapterModel.kt]: Please add `TODO` or remove commented out code

File: cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/AdapterModel.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    forEach {
+        items.addAll(it.value.toDisplayModel().items)
+    }
+
+    return DisplayModel(items)
+}
+
+internal fun CockpitParamGroup.toDisplayModel(addTopNavigation: Boolean = false): DisplayModel {
+    val items = mutableListOf<DisplayItem>(DisplayItem.Section(displayName))
+//    if (addTopNavigation) {
```
Comment (martawoldanska) [inline cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/AdapterModel.kt]: Same here

File: cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/ParamAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            ParamType.ACTION.ordinal -> ActionParamViewHolder(inflateViewForHolder(R.layout.cockpit_action_param, parent)).configure()
+            ParamType.COLOR.ordinal -> ColorParamViewHolder(inflateViewForHolder(R.layout.cockpit_color_param, parent)).configure()
+            ParamType.RANGE_INT.ordinal -> RangeIntParamViewHolder(inflateViewForHolder(R.layout.cockpit_range_param, parent)).configure()
+            ParamType.RANGE_DOUBLE.ordinal -> RangeDoubleParamViewHolder(inflateViewForHolder(R.layout.cockpit_range_param, parent)).configure()
+            ParamType.READ_ONLY.ordinal -> ReadOnlyParamViewHolder(inflateViewForHolder(R.layout.cockpit_read_only_param, parent)).configure()
+            ParamType.STEP_INT.ordinal -> StepIntParamViewHolder(inflateViewForHolder(R.layout.cockpit_step_param, parent)).configure()
+            ParamType.STEP_DOUBLE.ordinal -> StepDoubleParamViewHolder(inflateViewForHolder(R.layout.cockpit_step_param, parent)).configure()
+            GROUP_TYPE_ID -> GroupViewHolder(inflateViewForHolder(R.layout.subgroup, parent)).configure()
+            else -> SectionViewHolder(inflateViewForHolder(R.layout.cockpit_group_name, parent))
```
Comment (martawoldanska) [inline cockpit/src/main/java/com/polidea/cockpit/paramsedition/refactor/ParamAdapter.kt]: Please add separate case for section and throw an exception in `else`.

**Pull request merged**